tasks/internal_reconcile_report_orphan/src/adapters/api/aws.py

```diff
@@ -0,0 +1,166 @@
+    return {
+        "errorType": error_type,
+        "httpStatus": http_status_code,
+        "requestId": request_id,
+        "message": message,
+    }
+
+
+OS_ENVIRON_DB_CONNECT_INFO_SECRET_ARN_KEY = "DB_CONNECT_INFO_SECRET_ARN"  # nosec
+
+
+def handler(
+        event: Dict[str, Union[str, int]], context: Any
+) -> Union[List[Dict[str, Any]], Dict[str, Any]]:
+    """
+    Entry point for the internal_reconcile_report_orphan Lambda.
+    Args:
+        event: See schemas/input.json for details.
+        context: An object provided by AWS Lambda. Used for context tracking.
+
+    Environment Vars:
+        DB_CONNECT_INFO_SECRET_ARN (string):
+            Secret ARN of the AWS secretsmanager secret for connecting to the database.
+            See shared_db.py's get_configuration for further details.
+
+    Returns:
+        See schemas/output.json
+        Or, if an error occurs, see create_http_error_dict
+            400 if input does not match schemas/input.json.
+            500 if an error occurs when querying the database.
+    """
+    try:
+        LOGGER.setMetadata(event, context)
+
+        try:
+            _VALIDATE_INPUT(event)
+        except JsonSchemaException as json_schema_exception:
+            return src.adapters.api.aws.create_http_error_dict(
+                "BadRequest",
+                HTTPStatus.BAD_REQUEST,
+                context.aws_request_id,
+                json_schema_exception.__str__(),
+            )
+
+        db_connect_info = orca_shared.database.adapters.api.aws.get_configuration(
+            check_env_variable(OS_ENVIRON_DB_CONNECT_INFO_SECRET_ARN_KEY),
+            LOGGER
+        )
+
+        storage_adapter = src.adapters.storage.rdbms.StorageAdapterPostgres(
+            orca_shared.database.use_cases.create_postgres_connection_uri.create_user_uri(
+                db_connect_info, LOGGER))
+
+        orphan_record_filter = \
+            OrphanRecordFilter(job_id=event[INPUT_JOB_ID_KEY],
+                               page_index=event[INPUT_PAGE_INDEX_KEY],
+                               page_size=PAGE_SIZE)
+        orphan_record_page = src.use_cases.get_orphans_page.task(
+            orphan_record_filter,
+            storage_adapter,
+            LOGGER
+        )
+        result = {
+            OUTPUT_JOB_ID_KEY: event[INPUT_JOB_ID_KEY],
+            OUTPUT_ORPHANS_KEY: [{
+                ORPHANS_KEY_PATH_KEY: orphan.key_path,
+                ORPHANS_S3_ETAG_KEY: orphan.etag,
+                ORPHANS_S3_LAST_UPDATE_KEY: orphan.last_update,
+                ORPHANS_S3_SIZE_IN_BYTES_KEY: orphan.size_in_bytes,
+                ORPHANS_STORAGE_CLASS_KEY: orphan.storage_class,
+            } for orphan in orphan_record_page.orphans],
+            OUTPUT_ANOTHER_PAGE_KEY: orphan_record_page.another_page,
+        }
+
+        _VALIDATE_OUTPUT(result)
+
+        return result
+    except Exception as error:
+        return src.adapters.api.aws.create_http_error_dict(
```

Comment:
Any reason behind the full path here since `create_http_error_dict` is part of this file? I am thinking you can just call `create_http_error_dict`.

Reply:
I had an oddly difficult time getting that to work with unit tests.